components/backup-stream/src/endpoint.rs

```diff
@@ -953,6 +972,34 @@ where
                     }
                 });
             }
+            RegionCheckpointOperation::PrepareMinTsForResolve => {
+                let min_ts = self.pool.block_on(self.prepare_min_ts());
+                let start_time = Instant::now();
+                try_send!(
+                    self.scheduler,
+                    Task::RegionCheckpointsOp(RegionCheckpointOperation::Resolve {
+                        min_ts,
+                        start_time
+                    })
+                );
+            }
+            RegionCheckpointOperation::Resolve { min_ts, start_time } => {
+                let sched = self.scheduler.clone();
+                try_send!(
+                    self.scheduler,
+                    Task::ModifyObserve(ObserveOp::ResolveRegions {
+                        callback: Box::new(move |mut resolved| {
+                            let t =
+                                Task::RegionCheckpointsOp(RegionCheckpointOperation::Resolved {
+                                    checkpoints: resolved.take_resolve_result(),
+                                    start_time,
+                                });
+                            try_send!(sched, t);
+                        }),
+                        min_ts
+                    })
+                );
+            }
```

Comment:
It seems that `RegionCheckpointOperation::Resolve` is only scheduled by `RegionCheckpointOperation::PrepareMinTsForResolve` ?
```suggestion
            RegionCheckpointOperation::PrepareMinTsForResolve => {
                let min_ts = self.pool.block_on(self.prepare_min_ts());
                let start_time = Instant::now();
                let sched = self.scheduler.clone();
                try_send!(
                    self.scheduler,
                    Task::ModifyObserve(ObserveOp::ResolveRegions {
                        callback: Box::new(move |mut resolved| {
                            let t =
                                Task::RegionCheckpointsOp(RegionCheckpointOperation::Resolved {
                                    checkpoints: resolved.take_resolve_result(),
                                    start_time,
                                });
                            try_send!(sched, t);
                        }),
                        min_ts
                    })
                );
            }
```

Reply:
We must reschedule the task to consume all the pending events after we updated the concurrency manager.

Reply:
got it